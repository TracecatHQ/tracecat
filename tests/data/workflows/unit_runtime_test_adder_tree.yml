title: Adder tree Workflow with runtime tests
#    A
#    /\
#   B  c
description: |
  Tests correctness, templates + type casting, context passing, tests.
  B gets overridden by a test, but C's override is disabled.
entrypoint: a
actions:
  - ref: a
    action: core.transform.forward
    args:
      value: "1"

  - ref: b
    action: example.add
    depends_on:
      - a
    args:
      # Demonstrate casting
      lhs: ${{ ACTIONS.a.result -> int }}
      rhs: 1

  - ref: c
    action: example.add
    depends_on:
      - a
    args:
      lhs: 3
      rhs: ${{ ACTIONS.a.result -> int }}

tests:
  - ref: b
    validate_args: true
    success: 999

  - ref: c
    validate_args: true
    success: 1000

config:
  enable_runtime_tests: true
