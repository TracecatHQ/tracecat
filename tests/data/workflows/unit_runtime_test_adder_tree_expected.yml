ACTIONS:
  a:
    result: "1"
    result_typename: "str"
  b:
    result: 999
    result_typename: "int"
  c:
    result: 1000
    result_typename: "int"
INPUTS:

TRIGGER:
